joomla/joomla-cms
# [4.0] Fixing ordering of maps in smart search

Pull Request for Issue #30965.

### Summary of Changes
The maps view is not properly ordering the content, which is mainly because the table has been changed from a flat list to a nested table. Since the rows aren't ordered correctly, tasks like publishing the content are failing. This in turn is due to the code checking if parents are published and if the list of items to publish is not in the right order (parents first), that check fails and the whole process fails. This fixes the ordering (more or less) and thus hopefully fixes the publishing issue for now.


### Testing Instructions
1. Install Joomla, install some sampledata.
2. Go to Smart Search -> Maps
3. See that the rows are somehow strange, messed up, chaotic.
4. Select all maps and unpublish and then publish again. The later should fail.
5. Apply patch
6. See that for example authors show all up below "Authors"
7. Select all maps and click on publishing. Now this works.
